src/VisualStudio/Core/Def/Implementation/ProjectSystem/RuleSets/VisualStudioRuleSetManager.RuleSetFile.cs

```diff
@@ -154,8 +176,9 @@ private void IncludeUpdated(object sender, EventArgs e)
 
             private void IncludeUpdateCore()
             {
-                _ruleSetManager.StopTrackingRuleSetFile(this);
-                UnsubscribeFromFileTrackers();
+                // It's critical that Dispose() is called first prior to raising the event -- this way any callers who call the RuleSetManager
+                // asking for the new file are guaranteed to get the new snapshot first.
+                Dispose();
```

Comment:
As long as Dispose is idempotent, it's fine, which it's normally supposed to be anyways. I guess there's a small bug here because we might end up doing StopTrackingRuleSetFile more than once in that case. Will fix.